manim/scene/scene.py

```diff
@@ -84,7 +81,7 @@ def __init__(
         camera_class=Camera,
         always_update_mobjects=False,
         random_seed=None,
-        **kwargs,
```

Comment:
only remove kwargs if you are sure this will not cause crashes when we compile our code or our docs.

Otherwise, keep it here and add to the body something like 
```
if kwargs:
   logger.debug("Warning: Scene.__init__ received extra kwargs: %s", kwargs)
```

Reply:
I disagree. If extraneous keyword arguments are passed to something, a `TypeError` will be raised, which is correct behavior, because otherwise you're passing something that shouldn't be used. This is my biggest annoyance with `Container`. You can just pass any argument willy nilly and it doesn't even tell you unless you change the loglevel in the config

Reply:
Search for `TypeError: __init__() got an unexpected keyword argument 'XXX'` in the output.

Reply:
I don't see what you mean. The tests are currently failing because kwargs aren't being forwarded up to `Mobject` from `VMobject`

Reply:
So I presume the expected behavior is to unpack the keyword argument, and pass it in directly to the parent class?

Reply:
In that case, that should be investigated and fixed so that extraneous keyword arguments that don't do anything are removed. It doesn't make sense to allow them

Reply:
It seems this is a bigger refactor than I first thought; I'll continue to go through the TypeErrors/extraneous kwargs that are being thrown around. Should I mark this as a draft?

Reply:
For the record, I do not think you should do kwargs.pop for a keyword argument that you wish to pass to the parent. Each init function should process its own keyword arguments directly.

We have also been trying to make all of them explicit in the function signatures - so that we never need to use kwargs.pop at all because the function signature parsing will remove the listed keywords from the rest of the dictionary now named "kwargs".